### PR TITLE
[prometheus-node-exporter] add servicemonitor/podmonitor scrape limits

### PR DESCRIPTION
**Pull request by mblaschke**
Signed-off-by: Markus Blaschke <mblaschke82@gmail.com>

#### What this PR does / why we need it
adds support for servicemonitor/podmonitor scrape limits (eg. sampleLimit) to prometheus-node-exporter

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-node-exporter/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ keywords:
   - prometheus
   - exporter
 type: application
-version: 4.4.1
+version: 4.4.2
```
Comment (monotek) [inline charts/prometheus-node-exporter/Chart.yaml:9]: new feature shuld raise minor version

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ metadata:
   {{- end }}
 spec:
   jobLabel: {{ default "app.kubernetes.io/name" .Values.prometheus.monitor.jobLabel }}
+{{ include "servicemonitor.scrapeLimits" .Values.prometheus.monitor | indent 2 }}
```
Comment (monotek) [inline charts/prometheus-node-exporter/templates/servicemonitor.yaml:13]: indent looks wrong

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml (hunk 2 above)
Comment (mblaschke) [reply charts/prometheus-node-exporter/templates/servicemonitor.yaml:13]: huh? it's on the same level like `jobLabel`?

File: charts/prometheus-node-exporter/Chart.yaml (hunk 1 above)
Comment (zanhsieh) [reply charts/prometheus-node-exporter/Chart.yaml:9]:
```suggestion
version: 4.5.0
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml (hunk 2 above)
Comment (monotek) [reply charts/prometheus-node-exporter/templates/servicemonitor.yaml:13]: yes, its part of spec: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.ServiceMonitorSpec

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by mblaschke** [comment]
thanks, merged the changes

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -10,6 +10,7 @@ metadata:
   {{- end }}
 spec:
   jobLabel: {{ default "app.kubernetes.io/name" .Values.prometheus.monitor.jobLabel }}
+  {{- include "servicemonitor.scrapeLimits" .Values.prometheus.monitor | nindent 2 }}
```
Comment (monotek) [inline charts/prometheus-node-exporter/templates/servicemonitor.yaml:13]:
```suggestion
  {{- include "servicemonitor.scrapeLimits" .Values.prometheus.monitor | indent 2 }}
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/templates/servicemonitor.yaml (hunk 3 above)
Comment (mblaschke) [reply charts/prometheus-node-exporter/templates/servicemonitor.yaml]: fixed

**Review by zanhsieh — DISMISSED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**